pennylane/ops/qubit/parametric_ops_multi_qubit.py

```diff
@@ -325,6 +340,40 @@ def _check_pauli_word(pauli_word):
         """
         return all(pauli in PauliRot._ALLOWED_CHARACTERS for pauli in set(pauli_word))
 
+    def _check_word_batching(self, pauli_word):
+        """Check that the broadcasting/batching of a Pauli word is valid and
+        consistent with potential broadcasting of a ``PauliRot`` instance."""
```

Comment:
```suggestion
        """Check that the broadcasting/batching of a Pauli word is valid and
        consistent with potential broadcasting of a ``PauliRot`` instance.
        
        Side Effects:
            sets the `_batch_size` property.
        
        Raises:
            ValueError: if the pauli_word and theta batch sizes do not match or the pauli word is invalid.
        """
```

Not something we currently do much in PL, but something I really want to encourage going forward.